test/test_modes.jl

```diff
@@ -143,6 +143,27 @@ Emn2 = Eωm2/norm(Eωm2)
 @test all(isapprox.(En2, Emn2, atol=1e-3*maximum(abs.(Emn2))))
 end
 
+##
+@testset "overlap with Gaussian beam (function)" begin
```

Comment:
There is a testset in `test_fields.jl` that covers the non-diverged case.

Reply:
I'd forgotten about that. Will move this one there